### PR TITLE
added caching for call in lseries.py

### PR DESCRIPTION
**Pull request by kevinywlui**

### CODE REVIEW

File: sage_modabvar/lseries.py [hunk 1]
```diff
@@ -104,20 +104,26 @@ def __call__(self, s):
...
-            newforms = [Newform(simple.newform_label(),'a')
-                    for simple in simples]
-
-        factors = [newform.lseries(embedding=i)
-                for newform in newforms
-                for i in range(newform.base_ring().degree())]
-        return prod(L(s) for L in factors)
+        try:
+            factors = self.__factors
+            return prod(L(s) for L in factors)
```
Comment (williamstein) [inline sage_modabvar/lseries.py]: delete this line

File: sage_modabvar/lseries.py [hunk 2]
```diff
@@ -104,20 +104,26 @@ def __call__(self, s):
...
+            else:
+                simples = abelian_variety.decomposition()
+                newforms = [Newform(simple.newform_label(), 'a')
+                        for simple in simples]
+
+            factors = [newform.lseries(embedding=i)
+                    for newform in newforms
+                    for i in range(newform.base_ring().degree())]
+            self.__factors = factors
+            return prod(L(s) for L in factors)
```
Comment (williamstein) [inline sage_modabvar/lseries.py]: unindent this line.  

Then you have eliminated one line of code that was repeated twice. If you ever change it or find a bug, it's less likely that you'll have to fix the other copy of that line.

**Comment by williamstein** [comment]
I made a trivial comment; after you fix that, code looks fine.

**Comment by kevinywlui** [comment]
Fixed.

**Pull request merged**